wicket/src/state/inventory.rs

```diff
@@ -204,22 +201,13 @@ impl Component {
     // We also use this for the bootloader selection as the SIGN
     // of the bootloader is going to be identical to the RoT.
     pub fn rot_sign(&self) -> Option<Vec<u8>> {
-        match self.rot_active_slot() {
-            None => return None,
-            Some(s) => match s {
-                RotSlot::A => caboose_sign(
-                    self.sp()
-                        .rot
-                        .as_ref()
-                        .and_then(|rot| rot.caboose_a.as_ref()),
-                ),
-                RotSlot::B => caboose_sign(
-                    self.sp()
-                        .rot
-                        .as_ref()
-                        .and_then(|rot| rot.caboose_b.as_ref()),
-                ),
-            },
+        match self.rot_active_slot()? {
+            RotSlot::A => self.sp().rot.as_ref().map_or(None, |rot| {
+                rot.caboose_a.as_ref().map_or(None, |x| caboose_sign(x))
```

Comment:
I think this (and the one on line 209) could be

```suggestion
                rot.caboose_a.as_ref().and_then(caboose_sign)
```

which is a little clearer? `map_or` always trips me up.